### ISSUE TITLE
[Feature Request] Virtual midnight

### ISSUE THREAD

**Issue by asashnov** [gnome-pomodoro/gnome-pomodoro#686]
## Is your proposal related to a problem?

Programmers tend to work over midnight, going to bed around 2 a.m.
So it's more right to count some hours from 0:00 a.m. to 6 a.m. as a continuation of the current day.

## Describe the solution you'd like

For example, Hamster has option: 'New day starts at:"  (for ex., 6:00 a.m.)
(Ubuntu 22.04 hamster-time-tracker-3.0.2-3 package)

## Describe alternatives you've considered

I think it's possible to run gnome shell with different time, but haven't tried it, only theoretical:

```
faketime -f '-6h'  gnome-shell --replace &
```

## Other information

In other aspects, I find this program very cool: works, and lightweight.

**Comment by kamilprusko** [comment]
I agree. I was thinking about 4AM as the threshold as it's between 2AM and 5-6AM.

The stats are aggregated daily to make browsing stats snappier. Aggregation is done using SQL triggers, I don't think it can be easily implemented as a setting, it could be hard-coded though.

**Comment by asashnov** [comment]
Hard-coded midnight at 4AM should work for most, I think.

**Comment by kamilprusko** [comment]
It's implemented in`wip/gtk4` branch, though revamped stats are still in progress.

I've hard-coded it to 4AM.

If implemented as a setting, a simple SQL script won't be enough. It would need to be done in Vala as a background task to handle timezone/DST changes. It's a matter of erasing existing entries near the old midnight and rebuilding from history (now it's collected).
